src/fmelt.c

```diff
@@ -302,6 +303,7 @@ static void preprocess(SEXP DT, SEXP id, SEXP measure, SEXP varnames, SEXP valna
   SEXPTYPE type;
   data->lmax = 0; data->totlen = 0; data->nrow = length(VECTOR_ELT(DT, 0));
   SET_VECTOR_ELT(data->RCHK, 0, vars = checkVars(DT, id, measure, verbose));
+  data->measure_is_list = !isNull(measure) && isNewList(measure);
```

Comment:
huh, somewhat surprised we need the `!isNull()` test in addition to `isNewList()`, but it's right there in the definition of `isNewList()`:

https://github.com/r-devel/r-svn/blob/898b57332801c2ee3b26a3e41fea40aabe6a3fa2/src/include/Rinlinedfuns.h#L829-L832

```suggestion
  data->measure_is_list = !isNull(measure) && isNewList(measure); // NB: NULL passes isNewList() hence !isNull() too
```

Reply:
right